pumps/splunk.go

```diff
@@ -144,29 +146,73 @@ func (p *SplunkPump) WriteData(ctx context.Context, data []interface{}) error {
 	}).Info("Writing ", len(data), " records")
 	for _, v := range data {
 		decoded := v.(analytics.AnalyticsRecord)
-		apiKey := decoded.APIKey
-		if p.config.ObfuscateAPIKeys {
-			if len(apiKey) > 4 {
-				apiKey = "****" + apiKey[len(apiKey)-4:]
-			}
+		mapping := map[string]interface{}{
+			"method":         decoded.Method,
+			"host":           decoded.Host,
+			"path":           decoded.Path,
+			"raw_path":       decoded.RawPath,
+			"content_length": decoded.ContentLength,
+			"user_agent":     decoded.UserAgent,
+			"response_code":  decoded.ResponseCode,
+			"api_key":        decoded.APIKey,
+			"time_stamp":     decoded.TimeStamp,
+			"api_version":    decoded.APIVersion,
+			"api_name":       decoded.APIName,
+			"api_id":         decoded.APIID,
+			"org_id":         decoded.OrgID,
+			"oauth_id":       decoded.OauthID,
+			"raw_request":    decoded.RawRequest,
+			"request_time":   decoded.RequestTime,
+			"raw_response":   decoded.RawResponse,
+			"ip_address":     decoded.IPAddress,
+			"geo":            decoded.Geo,
+			"alias":          decoded.Alias,
 		}
 
-		event := map[string]interface{}{
-			"method":        decoded.Method,
-			"path":          decoded.Path,
-			"response_code": decoded.ResponseCode,
-			"api_key":       apiKey,
-			"time_stamp":    decoded.TimeStamp,
-			"api_version":   decoded.APIVersion,
-			"api_name":      decoded.APIName,
-			"api_id":        decoded.APIID,
-			"org_id":        decoded.OrgID,
-			"oauth_id":      decoded.OauthID,
-			"raw_request":   decoded.RawRequest,
-			"request_time":  decoded.RequestTime,
-			"raw_response":  decoded.RawResponse,
-			"ip_address":    decoded.IPAddress,
+		// Define an empty event
+		event := make(map[string]interface{})
+
+		// Populate the Splunk event with the fields set in the config
+		if len(p.config.Fields) > 0 {
+			// Loop through all fields set in the pump config
+			for _, field := range p.config.Fields {
+				// Skip the next actions in case the configured field doesn't exist
+				if _, ok := mapping[field]; !ok {
+					continue
+				}
+
+				// Check if the field is "api_key" and the obfuscation is configured
+				if field == "api_key" && p.config.ObfuscateAPIKeys {
```

Comment:
@mbelchev, in this way, we're obfuscating the key only if `api_key` is in the `fields` array. I would do it before creating the `mapping` map, so both config options are independent of each other. 
What do you think?

Reply:
@tbuchaillot, I was thinking about it but also considered that there might be a future improvement that will give the end-user the option to customize all fields on the fly with some kind of hooks. But for now, I agree with you and the change will be applied. Thanks for your feedback!